src/toast/observation_dist.py

```diff
@@ -73,6 +77,9 @@ def __init__(
         if self.comm is not None:
             self.comm_size = self.comm.size
             self.comm_rank = self.comm.rank
+            # Duplicate the input communicator, to ensure that we keep a copy if the
```

Comment:
Any risk of running out of communicators? I have seen such error before.

Reply:
From various online posts, it looks like this is (still) very implementation dependent.  However, I'm sure there is some overhead associated with each communicator.  I'll revert this.